tests/Unit/DataStructures/DataBox/Test_DataBox.cpp

```diff
@@ -1635,6 +1635,185 @@ SPECTRE_TEST_CASE("Unit.DataStructures.DataBox.Subitems",
+        Scalar<DataVector>(DataVector(2, 48.)));
+  CHECK(db::get<test_databox_tags::ScalarTag3>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 24.)));
+  CHECK(db::get<test_databox_tags::VectorTag3>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 10.))));
+  CHECK(db::get<test_databox_tags::ScalarTag4>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 12.)));
+  CHECK(db::get<test_databox_tags::VectorTag4>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 6.))));
+  {
+    const auto& vars =
+        db::get<test_databox_tags::MultiplyVariablesByTwo>(original_box);
+    CHECK(get<test_databox_tags::ScalarTag4>(vars) ==
+          Scalar<DataVector>(DataVector(2, 12.)));
+    CHECK(get<test_databox_tags::VectorTag4>(vars) ==
+          (tnsr::I<DataVector, 3>(DataVector(2, 6.))));
+  }
+  check_0(copied_box);
+
+  db::mutate<Tags::Variables<
+      tmpl::list<test_databox_tags::ScalarTag, test_databox_tags::VectorTag>>>(
+      make_not_null(&original_box), [](const auto vars) {
+        get<test_databox_tags::ScalarTag>(*vars).get() = 4.0;
+        get<test_databox_tags::VectorTag>(*vars) =
+            tnsr::I<DataVector, 3>(DataVector(2, 6.));
+      });
+
+  CHECK(db::get<test_databox_tags::ScalarTag>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 4.)));
+  CHECK(db::get<test_databox_tags::VectorTag>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 6.))));
+  CHECK(db::get<test_databox_tags::ScalarTag2>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 8.)));
+  CHECK(db::get<test_databox_tags::VectorTag2>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 2.))));
+  CHECK(db::get<test_databox_tags::MultiplyScalarByFour>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 32.)));
+  CHECK(db::get<test_databox_tags::MultiplyScalarByThree>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 96.)));
+  CHECK(db::get<test_databox_tags::DivideScalarByThree>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 32.)));
+  CHECK(db::get<test_databox_tags::ScalarTag3>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 16.)));
+  CHECK(db::get<test_databox_tags::VectorTag3>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 10.))));
+  CHECK(db::get<test_databox_tags::ScalarTag4>(original_box) ==
+        Scalar<DataVector>(DataVector(2, 8.)));
+  CHECK(db::get<test_databox_tags::VectorTag4>(original_box) ==
+        (tnsr::I<DataVector, 3>(DataVector(2, 12.))));
+  {
+    const auto& vars =
+        db::get<test_databox_tags::MultiplyVariablesByTwo>(original_box);
+    CHECK(get<test_databox_tags::ScalarTag4>(vars) ==
+          Scalar<DataVector>(DataVector(2, 8.)));
+    CHECK(get<test_databox_tags::VectorTag4>(vars) ==
+          (tnsr::I<DataVector, 3>(DataVector(2, 12.))));
+  }
+  check_0(copied_box);
+}
+
+// [[OutputRegex, Cannot make a deep copy of a Deferred that holds a
+// non-copyable data member.]]
```

Comment:
This is a user-facing error, so can it say what is wrong with the operation in terms of the DataBox instead of in terms of Deferred, which most users won't have a clue about?  I wouldn't understand this error message if I hadn't just reviewed the code causing it, and I'd still have to read the code again to figure out which of the two items in the box it was complaining about.

Reply:
100000% agreed :)